pro/src/components/RouteLeavingGuard/RouteLeavingGuard.tsx

```diff
@@ -68,6 +90,30 @@ const RouteLeavingGuard = ({
       )
   }
 
+  const rightButtonAction = () => {
+    if (typeof rightButton.action === 'function') {
+      return () => {
+        rightButton.action && rightButton.action()
```

Comment:
Tu ne devrais pas avoir besoin de `rightButton.action` vu que ligne 94, tu vérifies que c'est une fonction, non ?

Reply:
typescript râle si je rajoute pas ça...

`Cannot invoke an object which is possibly 'undefined'`